modules/standard/Regex.chpl

```diff
@@ -833,27 +837,29 @@ record regex {
   pragma "no doc"
   proc _search_match(text: exprType, anchor: c_int, param has_captures, ref captures):regexMatch
   {
+    /* This pattern is a bit ugly, but we'd like to avoid
+    unnecessarily copying `this` when local. And we need to maintain
+    the lifetime of the copied regex when remote. */
+    var regexCopy:regex(exprType);
+    if home != here then regexCopy = this;
+    const localRegex = if home != here then regexCopy._regex else _regex;
     var ret:regexMatch;
-    on this.home {
-      var pos:byteIndex;
-      var endpos:byteIndex;
-
-      pos = 0;
-      endpos = pos + text.numBytes;
-
-      var matches:_ddata(qio_regex_string_piece_t);
-      param nmatches = if has_captures then 1 + captures.size else 1;
-      matches = _ddata_allocate(qio_regex_string_piece_t, nmatches);
-      var got:bool;
-      got = qio_regex_match(_regex, text.localize().c_str(), text.numBytes,
-                             pos:int, endpos:int, anchor,
-                             matches, nmatches);
-      // Now try to coerce the read strings into the captures.
-      if has_captures then _handle_captures(text, matches, nmatches, captures);
-      // Now return where we matched.
-      ret = new regexMatch(got, matches[0].offset:byteIndex, matches[0].len);
-      _ddata_free(matches, nmatches);
-    }
+    var pos:byteIndex;
+    var endpos:byteIndex;
+
+    pos = 0;
+    endpos = pos + text.numBytes;
+
+    param nmatches = if has_captures then 1 + captures.size else 1;
+    var matches = new c_array(qio_regex_string_piece_t, nmatches);
```

Comment:
Could alternatively be written as `var matches: c_array(qio_regex_string_piece_t, nmatches);` which I guess is more what I would expect since you're really default initializing (the arguments are type information). Not a big deal though.

Reply:
Yeah looking at other modules, they all use the form you suggested, changing now